dotnet/src/Microsoft.Agents.AI.Mem0/Mem0Client.cs

```diff
@@ -0,0 +1,177 @@
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>Enumerable of memory strings.</returns>
+    public async Task<IEnumerable<string>> SearchAsync(string? applicationId, string? agentId, string? threadId, string? userId, string? inputText, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrWhiteSpace(applicationId)
+            && string.IsNullOrWhiteSpace(agentId)
+            && string.IsNullOrWhiteSpace(threadId)
+            && string.IsNullOrWhiteSpace(userId))
+        {
+            throw new ArgumentException("At least one of applicationId, agentId, threadId, or userId must be provided.");
+        }
+
+        var searchRequest = new SearchRequest
+        {
+            AppId = applicationId,
+            AgentId = agentId,
+            RunId = threadId,
+            UserId = userId,
+            Query = inputText ?? string.Empty
+        };
+
+        using var content = new StringContent(JsonSerializer.Serialize(searchRequest, Mem0SourceGenerationContext.Default.SearchRequest), Encoding.UTF8, "application/json");
+        using var responseMessage = await this._httpClient.PostAsync(s_searchUri, content, cancellationToken).ConfigureAwait(false);
+        responseMessage.EnsureSuccessStatusCode();
+
+#pragma warning disable CA2016 // ReadAsStringAsync has no overload with token in current target; token already applied to request
+        var response = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
+#pragma warning restore CA2016
+        var searchResponseItems = JsonSerializer.Deserialize(response, Mem0SourceGenerationContext.Default.SearchResponseItemArray);
+        return searchResponseItems?.Select(item => item.Memory) ?? Array.Empty<string>();
+    }
+
+    /// <summary>
+    /// Creates a memory for the provided message content.
+    /// </summary>
+    public async Task CreateMemoryAsync(string? applicationId, string? agentId, string? threadId, string? userId, string messageContent, string messageRole, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrWhiteSpace(applicationId)
+            && string.IsNullOrWhiteSpace(agentId)
+            && string.IsNullOrWhiteSpace(threadId)
+            && string.IsNullOrWhiteSpace(userId))
+        {
+            throw new ArgumentException("At least one of applicationId, agentId, threadId, or userId must be provided.");
+        }
+
+#pragma warning disable CA1308 // Lowercase required by service
+        var createMemoryRequest = new CreateMemoryRequest
+        {
+            AppId = applicationId,
+            AgentId = agentId,
+            RunId = threadId,
+            UserId = userId,
+            Messages = new[]
+            {
+                new CreateMemoryMessage
+                {
+                    Content = messageContent,
+                    Role = messageRole.ToLowerInvariant()
+                }
+            }
+        };
+#pragma warning restore CA1308
```

Comment:
Similar here.

Update the `setter` to do the `lowercase` instead.

```suggestion
        var createMemoryRequest = new CreateMemoryRequest
        {
            AppId = applicationId,
            AgentId = agentId,
            RunId = threadId,
            UserId = userId,
            Messages = new[]
            {
                new CreateMemoryMessage
                {
                    Content = messageContent,
                    Role = messageRole
                }
            }
        };
```